blocker/blocker.go

```diff
@@ -94,6 +79,61 @@ func New(ctx context.Context, skydAPI skyd.API, db *database.DB, logger *logrus.
 	return bl, nil
 }
 
+// BlockHashes blocks the given list of hashes. It returns the amount of hashes
+// which were blocked successfully, the amount that were invalid, and a
+// potential error.
+func (bl *Blocker) BlockHashes(hashes []database.Hash) (int, int, error) {
+	start := 0
+
+	// we have to keep track of what hashes were blocked, what hashes were
+	// invalid and what hainvalid, and what hashes we
+	// failed to block, because we don't want to retry the invalid hashes
+	var numBlocked int
+	var numInvalid int
+
+	for start < len(hashes) {
+		// check whether we need to escape
+		select {
+		case <-bl.staticCtx.Done():
+			return numBlocked, numInvalid, nil
+		default:
+		}
+
+		// calculate the end of the batch range
+		end := start + BlockBatchSize
+		if end > len(hashes) {
+			end = len(hashes)
+		}
+
+		// create the batch
+		batch := hashes[start:end]
+
+		// send the batch to skyd, if an error occurs we mark it as failed and
+		// escape early because something is probably wrong
+		blocked, invalid, err := bl.staticSkydAPI.BlockHashes(batch)
+		if err != nil {
+			err = errors.Compose(err, bl.staticDB.MarkAsFailed(batch))
+			return numBlocked, numInvalid, err
+		}
+
+		// update the counts
+		numBlocked += len(blocked)
+		numInvalid += len(invalid)
+
+		// update the documents
+		err1 := bl.staticDB.MarkAsSucceeded(blocked)
```

Comment:
what do you propose? I personally don't mind the naming here

Reply:
Even `MarkAsNotFailed` would be better, as it won't imply the existence of a `succeeded` field or something of the sort.

Reply:
I think it's fine. Because it actually succeeded. Sure the field says `failed` but the point of the verbose setter is to move the argument into the method name as opposed to `MarkFailed(true || false)`. Which I would be fine with as well.

Although if I had to nitpick I'd probably prefer a shorter name like `MarkSucceeded` and `MarkFailed`.

Only nit I'd probably have is that it could be shortened to `MarkFailed` and `MarkSucceeded`.

Reply:
Ok, I'll update it to `MarkSucceeded` and `MarkFailed`. I would usually entertain the reviewer and update it but in this case, I just disagree that it's confusing. There's no `succeeded field because a boolean should not be defined by two fields in the database. `MarkSucceeded` is the right name because that's just what it does, _how_ it marks the document as succeeded is irrelevant. I owe you one Ivo :)